### PR TITLE
Disallows siphoning credits outside of station

### PR DESCRIPTION
**Pull request by Maia-J**
## About The Pull Request

See name
## Why It's Good For The Game

Imagine a funny little antagonist going to deep space with one of these. You'll never find them, and even if it were to have a gps signal by the time you were out to the location they could (And most like will) be long gone, or in a different ruin entirely.

Basically this is easy to abuse and not fun to play against.
## Changelog
:cl:
balance: Disallows siphoning credits outside of station
/:cl:

### CODE REVIEW

File: code/game/machinery/bank_machine.dm [hunk 1]
```diff
@@ -105,8 +105,11 @@
 
 	switch(action)
 		if("siphon")
-			say("Siphon of station credits has begun!")
-			start_siphon(ui.user)
+			if(is_station_level(src.z) || is_centcom_level(src.z))
```
Comment (Ghommie) [inline code/game/machinery/bank_machine.dm:108]: For consistency, do this for the "halt" action too.

**Comment by MrMelbert** [comment]
No measure to prevent people from starting a siphon on station z level and then resuming it off station z (whiteship and other shuttles)?

File: code/game/machinery/bank_machine.dm (hunk 1 above)
Comment (Majkl-J) [reply code/game/machinery/bank_machine.dm]: I'll go with what melbert said, sounds like a better idea

File: code/game/machinery/bank_machine.dm [hunk 2]
```diff
@@ -67,6 +67,10 @@
 		say("Insufficient power. Halting siphon.")
 		end_siphon()
 		return
+	if (!(is_station_level(src.z) || is_centcom_level(src.z)))
+		say("Error: Console not in reach of station. Siphon halted.")
+		end_siphon()
+		return
```
Comment (MrMelbert) [inline code/game/machinery/bank_machine.dm:73]: This is not quite what I said
You should still prevent people from enabling siphoning while off Z, you just need to implement `on_changed_z_level` -> `if not valid z, end_siphon`

**Pull request merged**